src/views/Deposit/DepositConfirm.tsx

```diff
@@ -295,11 +294,13 @@ const DepositConfirm: React.FC<{}> = ({}) => {
     },
     {
       onSuccess: async (unitAmountResult, vars, context) => {
+        console.log("approvalMutation:onSuccess");
         await Promise.allSettled([
           // queryClient.invalidateQueries(approvedQueryKey), // Request that the approval query refreshes
           queryClient.setQueryData(approvedQueryKey, ethers.utils.parseEther(wholeTokenAmount.toString())), // Update the approved amount query immediately
           queryClient.invalidateQueries(approvalMutationKey),
         ]);
+        setStep(2);
```

Comment:
Mutations should alter the frontend or the backend, but never both.
Use mutateAsync instead of mutate at the call site, allowing you to use mutateAsync.then((result) => { (on success things); });

Reply:
Refactor this in the future- merging now since it’s not urgent.